cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/upgrade/AbstractConfigurationUpgrader.java

```diff
@@ -39,335 +40,357 @@
+	 */
+	private String configurationXsdFilename;
+
+	/** JAXB root object class of the configuration for (un-)marshalling */
+	private Class<?> configurationJaxbRootNode;
+
+	/**
+	 * Package prefix generated JAXB classes of the configurations (analogous to the
+	 * pom.xml specification)
+	 */
+	private static final String JAXB_PACKAGE_PREFIX = "com.devonfw.cobigen.impl.config.entity.io";
+
+	/**
+	 * Creates a new instance for the abstract implementation of an configuration
+	 * upgrader.
+	 *
+	 * @param version                   an arbitrary instance to perform reflection
+	 *                                  on during upgrading.
+	 * @param configurationJaxbRootNode JAXB root object class of the configuration
+	 *                                  for (un-)marshalling.
+	 * @param configurationFilename     standardized file name of the configuration
+	 * @author mbrunnli (Jun 23, 2015)
+	 */
+	@SuppressWarnings("unchecked")
+	AbstractConfigurationUpgrader(VERSIONS_TYPE version, Class<?> configurationJaxbRootNode,
+			String configurationFilename) {
+
+		this.configurationJaxbRootNode = configurationJaxbRootNode;
+		this.configurationFilename = configurationFilename;
+
+		// split camel case
+		this.configurationName = StringUtils.join(
+				configurationJaxbRootNode.getSimpleName().split("(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=-Z][a-z])"), " ");
+
+		this.configurationXsdFilename = StringUtils.uncapitalize(configurationJaxbRootNode.getSimpleName()) + ".xsd";
+
+		// determine all "version enum" values
+		try {
+			this.versions = (VERSIONS_TYPE[]) version.getClass().getMethod("values").invoke(version);
+		} catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException
+				| SecurityException e) {
+			throw new CobiGenRuntimeException(
+					"Unexpected happend! Could not determine values of the enum '" + version + "'.", e);
+		}
+	}
+
+	/**
+	 * Checks, whether the configuration is compliant to the latest supported
+	 * version.
+	 *
+	 * @param configurationRoot the root folder containing the configuration
+	 * @return <code>true</code> if the configuration is up-to-date, <br>
+	 *         otherwise <code>false</code>
+	 */
+	public boolean isCompliantToLatestSupportedVersion(Path configurationRoot) {
+
+		return resolveLatestCompatibleSchemaVersion(configurationRoot, true) != null;
+	}
+
+	/**
+	 * Checks, whether the configuration can be read with an old schema version.
+	 *
+	 * @param configurationRoot the root folder containing the configuration
+	 * @return the newest schema version, the configuration is compliant with or
+	 *         <code>null</code> if the configuration is not compliant to any.
+	 */
+	public VERSIONS_TYPE resolveLatestCompatibleSchemaVersion(Path configurationRoot) {
+		VERSIONS_TYPE result = resolveLatestCompatibleSchemaVersion(configurationRoot, false);
+		// if CobiGen does not understand the upgraded file... throw exception
```

Comment:
Add TODO.

Reply:
This check will be still done in the upgradeConfigurationToLatestVersion and will be removed